pkg/distributor/distributor.go

```diff
@@ -1195,6 +1233,12 @@ func (d *Distributor) exceedsLimits(ctx context.Context, tenantID string, stream
 	// Send the request to the limits-frontend to see if it exceeds the tenant
 	// limits. If the RPC fails, failover to the next instance in the ring.
 	for _, instance := range rs.Instances {
+		select {
+		case <-ctx.Done():
```

Comment:
This checks for timeout to avoid calling each instance when `ctx` has already been canceled.